tracer/src/Datadog.Trace/Util/IdGenerator.cs

```diff
@@ -0,0 +1,36 @@
+// <copyright file="IdGenerator.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+namespace Datadog.Trace.Util;
+
+/// <summary>
+/// Generates random numbers suitable for use as Datadog trace ids and span ids.
+/// </summary>
+internal class IdGenerator
+{
+    /// <summary>
+    /// Returns a random number greater than zero and less than Int64.MaxValue.
+    /// </summary>
+    public static ulong NextUInt64()
+    {
+        while (true)
+        {
+#if NET6_0_OR_GREATER
+            // random integer in range [0, Int64.MaxValue)
+            var value = (ulong)System.Random.Shared.NextInt64();
+#else
+            // random integer in range [0, UInt64.MaxValue],
+            // clear highest bit to make it [0, Int64.MaxValue]
+            var value = RandomNumberGenerator.Current.NextUInt64() & 0x7FFFFFFFFFFFFFFF;
+#endif
+
+            if (value is > 0 and < long.MaxValue)
```

Comment:
Hmm, I think I confused this with the 2^63 limit, which is a different thing. Thanks!
(fwiw, Java and Go do seem to use `< Int64.MaxValue`, see [this internal doc](https://docs.google.com/spreadsheets/d/11_cKJ3BBLsVqLxZCkMgLPATtVfSLf5bFeVHOEVJhEVs/edit?usp=sharing))

Reply:
Hmm, I don't know if I can make this work for the .NET 6 implementation above:
```csharp
// random integer in range [0, Int64.MaxValue)
var value = (ulong)System.Random.Shared.NextInt64();
```

By default the max value returned is `Int64.MaxValue`, and even if I try to specify a value, the maximum is always exclusive (while the minimum is inclusive). I can't add 1 to `Int64.MaxValue` like this:
```csharp
var value = (ulong)System.Random.Shared.NextInt64(1, Int64.MaxValue + 1);
```

Reply:
Maybe I should use `RandomNumberGenerator.NextUInt64()` for all platforms, even .NET 6?

Reply:
TBH, I'm not following much. IMO, it should have returned any value from 0 to `long.MaxValue`. Even though I understand from all the other comments that it's not that easy. Anyway it isn't a game changer. If `Random` could return `long.MaxValue`, thus it would be just one extra iteration happening rather rarely. So we could also leave it like that.

Reply:
> By default the max value returned is `Int64.MaxValue`, and even if I try to specify a value, the maximum is always exclusive (while the minimum is inclusive). I can't add 1 to `Int64.MaxValue` like this:

IIUC: 
- `System.Random.Shared.NextInt64()` returns `[0, Int64.MaxValue)`
- We want to return `[1, Int64.MaxValue]` (inclusive both sides)
- We can rewrite what we want to return as `[1, Int64.MaxValue + 1)`
- Or `[0, Int64.MaxValue) + 1`
- So we can do `System.Random.Shared.NextInt64()` + 1, no?

Reply:
> By default the max value returned is `Int64.MaxValue`

Sorry, I meant the max value returned by `System.Random.NextInt64()` is `Int64.MaxValue - 1`, not `Int64.MaxValue`. I've edited my previous comment above.

> We want to return `[1, Int64.MaxValue]` (inclusive both sides)

I don't really care about the maximum including `Int64.MaxValue` or not. Some tracers do it, some don't. I'd say go with whatever is easier to implement in your runtime. Note that we do _not_ include `Int64.MaxValue` today since we rely on `System.Random`. (The document I linked above was wrong, I just edited it as well.)

 > So we can do `System.Random.Shared.NextInt64() + 1`, no?

Ugh, so easy, thanks. Let me try that.

Reply:
I made some changes, are we now able to return `long.MaxValue` without writing a special case for .NET 6+.